test/showcase/validate.js

```diff
@@ -1,21 +1,49 @@
-import {readdir} from 'node:fs/promises';
+import {readdirSync, statSync} from 'node:fs';
+import {basename} from 'node:path';
 
+import expect from 'expect.js';
+import imageSize from 'image-size';
 import validateSchema from 'yaml-schema-validator';
 
 import showcaseEntrySchema from './schema.js';
 
 
 const SHOWCASE_DIR = 'data/showcase';
+const ILLUSTRATIONS_DIR = 'static/img/showcase';
 
-const entries = await readdir(SHOWCASE_DIR);
+describe('Showcase entries', () => {
+  const entries = readdirSync(SHOWCASE_DIR);
 
-const errorCount = entries.map((entry) => {
-  console.log(entry);
-  return validateSchema(`${SHOWCASE_DIR}/${entry}`, {
-    schema: showcaseEntrySchema,
-  });
-}).reduce((errorCount, errors) => {
-  return errorCount + errors.length;
-}, 0);
+  entries.forEach((entry) => {
+    entry = basename(entry, '.yml');
+
+    describe(entry, () => {
+      it('has a valid description', () => {
+        const validationErrors = validateSchema(`${SHOWCASE_DIR}/${entry}.yml`, {
+          schema: showcaseEntrySchema,
+        });
+        expect(validationErrors).to.be.empty();
+      });
+
+      describe('illustration', () => {
+        const path = `${ILLUSTRATIONS_DIR}/${entry}.png`;
+        let stats;
 
-process.exitCode = errorCount;
+        it('exists', () => {
+          stats = statSync(path);
+        });
+
+        it('is not too heavy', () => {
+          expect(stats.size).to.be.below(400 * 1024); // 400 kiB
+        });
+
+        it('is large enough', () => {
+          const dimensions = imageSize(path);
+          expect(dimensions.width).to.be.greaterThan(1279);
+          expect(dimensions.height).to.be.greaterThan(719);
+          expect(dimensions.width / dimensions.height).to.be.within(1.7, 1.8); // expect a 16:9 ratio
```

Comment:
I'm not sure I understand the added value of allowing larger dimensions with the same ratio, I'd tend to simplify by specifying the mandatory width/height values.

```suggest
expect(dimensions.width).to.be(1280);
expect(dimension.height.to.be(720);
```

Reply:
My idea was to:

1. Enable higher quality pictures as long as they fulfil the weight constraint.
2. Make it easier for contributors to provide “good enough” pictures. I noticed how in the contributed illustrations I rarely received 1280x720 pictures even though that was the requirement 😅